### PR TITLE
dev -> Master for 2.0.0 release

### PR DESCRIPTION
**Pull request by Emiller88**

### CODE REVIEW

File: .github/workflows/branch.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      # PRs to the nf-core repo master branch are only ok if coming from the nf-core repo `dev` or any `patch` branches
+      - name: Check PRs
+        if: github.repository == 'nf-core/nascent'
+        run: |
+          { [[ ${{github.event.pull_request.head.repo.full_name }} == nf-core/nascent ]] && [[ $GITHUB_HEAD_REF = "dev" ]]; } || [[ $GITHUB_HEAD_REF == "patch" ]]
```
Comment (FriederikeHanssen) [inline .github/workflows/branch.yml:16]: I think there was something discussed on slack whether double quotes may be needed again, commenting so I won't forget to look this up....

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        NXF_VER:
+          - "21.10.6"
+          - "latest-everything"
+        parameters:
+          - "--transcript_identification homer"
+          - "--aligner dragmap --transcript_identification homer"
+          # - "--transcript_identification grohmm"
```
Comment (FriederikeHanssen) [inline .github/workflows/ci.yml:28]: no testing this param?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,30 @@
...
+
+### `Added`
+
+- [[#28](https://github.com/nf-core/nascent/issues/28)] - Added DRAGMAP alignment
+- [[#64](https://github.com/nf-core/nascent/pull/64)] - Added CHM13 igenomes config
+- [[#39](https://github.com/nf-core/nascent/issues/39)] - Add PINTS for TSS identification
+- [[#71](https://github.com/nf-core/nascent/issues/71)] - Add FASTP for adapter trimming
+- [[#77](https://github.com/nf-core/nascent/issues/77)] - Add dedup subworkflow
+
+### `Fixed`
```
Comment (FriederikeHanssen) [inline CHANGELOG.md:16]: no mention of DSL2 conversion?

File: LICENSE [hunk 4]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) Ignacio Tripodi, Margaret Gruca
+Copyright (c) Edmund Miller
```
Comment (FriederikeHanssen) [inline LICENSE:3]: not sure if authors should be removed, or just more added in this case (unless of course it is agreed with everyone) : https://nf-co.re/docs/contributing/guidelines/requirements/acknowledgements

File: README.md [hunk 5]
```diff
@@ -1,84 +1,93 @@
...
+## Introduction
 
+**nf-core/nascent** is a bioinformatics best-practice analysis pipeline for nascent transcript (NT) and Transcriptional Start Site (TSS) assays.
 
-#### Reference
+The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It uses Docker/Singularity containers making installation trivial and results highly reproducible. The [Nextflow DSL2](https://www.nextflow.io/docs/latest/dsl2.html) implementation of this pipeline uses one container per process which makes it much easier to maintain and update software dependencies. Where possible, these processes have been submitted to and installed from [nf-core/modules](https://github.com/nf-core/modules) in order to make them available to all nf-core pipelines, and to everyone within the Nextflow community!
 
-If you've used this pipeline in your research, you can cite this pipeline using DOI 10.17605/OSF.IO/SV4UB ([OSF project](https://osf.io/sv4ub/)).
+<!-- TODO nf-core: Add full-sized test dataset and amend the paragraph below if applicable -->
```
Comment (FriederikeHanssen) [inline README.md:19]: are the full size tests available?

File: README.md [hunk 6]
```diff
@@ -1,84 +1,93 @@
...
-5. [Troubleshooting](https://nf-co.re/usage/troubleshooting)
+On release, automated continuous integration tests run the pipeline on a full-sized dataset on the AWS cloud infrastructure. This ensures that the pipeline runs on AWS, has sensible resource allocation defaults set to run on real-world datasets, and permits the persistent storage of results to benchmark between pipeline releases and other analysis sources.The results obtained from the full-sized test can be viewed on the [nf-core website](https://nf-co.re/nascent/results).
 
-This pipeline is designed to process the sequencing output of nascent transcription assays, like GRO-seq or PRO-seq. It produces bedGraph- and bigWig-fomatted outputs after mapping strand-specific reads, as well as other useful outputs like quality control reports or IGV-ready (Integrative Genomics Viewer) tdf files.
+## Pipeline summary
 
-### Quick start
+1. Read QC ([`FastQC`](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/))
+2. Present QC for raw reads ([`MultiQC`](http://multiqc.info/))
+3. Transcript identification ([`GroHMM`](https://bioconductor.org/packages/release/bioc/html/groHMM.html))
```
Comment (FriederikeHanssen) [inline README.md:28]: anything else? Mapping, duplicate marking...?

File: README.md [hunk 7]
```diff
@@ -1,84 +1,93 @@
...
+
+## Contributions and Support
+
+If you would like to contribute to this pipeline, please see the [contributing guidelines](.github/CONTRIBUTING.md).
+
+For further information or help, don't hesitate to get in touch on the [Slack `#nascent` channel](https://nfcore.slack.com/channels/nascent) (you can join with [this invite](https://nf-co.re/join/slack)).
+
+## Citations
+
+<!-- TODO nf-core: Add citation for pipeline after first release. Uncomment lines below and update Zenodo doi and badge at the top of this file. -->
```
Comment (FriederikeHanssen) [inline README.md:82]: Since there is a previous release, there should be a zenodo link around that should be added here

File: conf/test_full.config [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+----------------------------------------------------------------------------------------
+*/
+
+params {
+    config_profile_name        = 'Full test profile'
+    config_profile_description = 'Full test dataset to check pipeline function'
+
+    // Input data for full size test
+    // TODO nf-core: Specify the paths to your full test data ( on nf-core/test-datasets or directly in repositories, e.g. SRA)
```
Comment (FriederikeHanssen) [inline conf/test_full.config:18]: can these todo statements be removed?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,4 +1,30 @@
 # nf-core/nascent: Changelog
 
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## v2.0.0 - 2022-10-18
```
Comment (apeltzer) [inline CHANGELOG.md:6]: No codename 😢

File: README.md [hunk 10]
```diff
@@ -1,84 +1,93 @@
-# nf-core/nascent
+# ![nf-core/nascent](docs/images/nf-core-nascent_logo_light.png#gh-light-mode-only) ![nf-core/nascent](docs/images/nf-core-nascent_logo_dark.png#gh-dark-mode-only)
 
-**Nascent Transcription Processing Pipeline**.
+[![AWS CI](https://img.shields.io/badge/CI%20tests-full%20size-FF9900?labelColor=000000&logo=Amazon%20AWS)](https://nf-co.re/nascent/results)[![Cite with Zenodo](http://img.shields.io/badge/DOI-10.5281/zenodo.XXXXXXX-1073c8?labelColor=000000)](https://doi.org/10.5281/zenodo.XXXXXXX)
 
-[![Build Status](https://travis-ci.com/nf-core/nascent.svg?branch=master)](https://travis-ci.com/nf-core/nascent)
-[![Nextflow](https://img.shields.io/badge/nextflow-%E2%89%A50.32.0-brightgreen.svg)](https://www.nextflow.io/)
-[![DOI](https://zenodo.org/badge/157735234.svg)](https://zenodo.org/badge/latestdoi/157735234)
```
Comment (apeltzer) [inline README.md:7]: This is the DOI for Nascent - should be added back :-) 

[DOI](https://zenodo.org/badge/latestdoi/157735234)

File: README.md (hunk 7 above)
Comment (apeltzer) [reply README.md:82]: [DOI](https://zenodo.org/badge/latestdoi/157735234)

File: assets/multiqc_config.yml [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+run_modules:
+  - custom_content
+  - fastqc
+  - fastp
+  - samtools
+  - preseq
+  - rseqc
+  - featureCounts
+  - homer
```
Comment (apeltzer) [inline assets/multiqc_config.yml:21]: If these tools run, should also maybe have them listed in the docs - which are currently a bit sparse :-)

File: tests/config/pytest_software.yml [hunk 12]
```diff
@@ -0,0 +1,5 @@
+# FIXME Tests don't work because of bin/ scripts
```
Comment (apeltzer) [inline tests/config/pytest_software.yml:1]: Is that still true?

File: tests/modules/local/grohmm/test.yml [hunk 13]
```diff
@@ -0,0 +1,27 @@
+- name: Run grohmm makeucscfile
+  command: nextflow run tests/modules/local/grohmm/ -entry test_grohmm_makeucscfile -c tests/config/nextflow.config -c tests/modules/local/grohmm/nextflow.config --transcript_identification grohmm
+  # FIXME Tests don't work because of bin/ scripts
```
Comment (apeltzer) [inline tests/modules/local/grohmm/test.yml:3]: Same as above?

File: assets/methods_description_template.yml [hunk 14]
```diff
@@ -0,0 +1,25 @@
+id: "nf-core-nascent-methods-description"
+description: "Suggested text and references to use when describing pipeline usage within the methods section of a publication."
+section_name: "nf-core/nascent Methods Description"
+section_href: "https://github.com/nf-core/nascent"
+plot_type: "html"
+## TODO nf-core: Update the HTML below to your prefered methods description, e.g. add publication citation for this pipeline
```
Comment (apeltzer) [inline assets/methods_description_template.yml:6]:
```suggestion

```

**Review by apeltzer — APPROVED** [review]
👍🏻 Minor comments - you have quite a bit of extra tests in the pipeline, which Ive not seen in others so far.

File: README.md (hunk 6 above)
Comment (Emiller88) [reply README.md:28]: Missed this one, it was all in output though :upside_down_face:

File: conf/test_full.config (hunk 8 above)
Comment (Emiller88) [reply conf/test_full.config:18]: This one is still relevant #78

File: assets/multiqc_config.yml (hunk 11 above)
Comment (Emiller88) [reply assets/multiqc_config.yml:21]: They're all in output, I missed updating the Pipeline summary, but that should be up-to-date now!

File: LICENSE (hunk 4 above)
Comment (Emiller88) [reply LICENSE:3]: Yeah, this got changed because it kept breaking the template branch. Happy to add it back, I wasn't sure what to do about it. I think doing this has brought up some good points about what happens when a pipeline picks up new maintainers or the maintainers move on.

File: assets/methods_description_template.yml (hunk 14 above)
Comment (Emiller88) [reply assets/methods_description_template.yml:6]: Leaving this as a reminder to add later, unless you think I should remove it.

File: tests/config/pytest_software.yml (hunk 12 above)
Comment (Emiller88) [reply tests/config/pytest_software.yml:1]: Just threw some stuff together, and yes it's still true, I think it needs to come from the nextflow level before we can start testing local modules that depend on bin/ scripts. Or make more heavy use of custom modules.

**Review by Emiller88 — COMMENTED** [review]
Thanks for checking so thoroughly so far!

File: tests/modules/local/grohmm/test.yml (hunk 13 above)
Comment (Emiller88) [reply tests/modules/local/grohmm/test.yml:3]: Yep!

File: tests/config/pytest_software.yml (hunk 12 above)
Comment (Emiller88) [reply tests/config/pytest_software.yml:1]: https://github.com/nf-core/nascent/commit/0ff751ad7de9f167d8bf794a2c044db7c88e16bb

**Comment by apeltzer** [comment]
Retriggered the tests, something odd (maybe network outage):

`ERROR: Please check input samplesheet -> Read 1 FastQ file does not exist!`

File: .github/workflows/branch.yml (hunk 1 above)
Comment (FriederikeHanssen) [reply .github/workflows/branch.yml:16]: hmmmm I don't see the branch protection test fail or pass. did you turn it of?

File: README.md (hunk 10 above)
Comment (FriederikeHanssen) [reply README.md:7]: there is a main doi so to say that points to the whole thing, and then one per release, we are usually using the main one for the badges as the don't change between releases 
Screenshot 2022-10-20 at 00 00 33

File: README.md (hunk 5 above)
Comment (FriederikeHanssen) [reply README.md:19]: ok, so yes, but with cool upgrades in the future :D

File: bin/parameter_tuning.R [hunk 15]
```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env Rscript
+# TODO
```
Comment (FriederikeHanssen) [inline bin/parameter_tuning.R:2]: anything left here TODO?

File: bin/parameter_tuning.R [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+# Allow for user to input their own annotation dataset
+# REQUIREMENTS
```
Comment (FriederikeHanssen) [inline bin/parameter_tuning.R:4]: I assume you are the author and no further licenses etc need to be added?

File: bin/transcriptcalling_grohmm.R [hunk 17]
```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env Rscript
+# TODO
```
Comment (FriederikeHanssen) [inline bin/transcriptcalling_grohmm.R:2]: TODO?

File: conf/igenomes.config [hunk 18]
```diff
@@ -1,146 +1,440 @@
...
+            star        = "${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/STARIndex/"
+            bismark     = "${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/BismarkIndex/"
+            gtf         = "${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Annotation/Genes/genes.gtf"
+            bed12       = "${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Annotation/Genes/genes.bed"
+            mito_name   = "chrM"
+            macs_gsize  = "2.7e9"
+            blacklist   = "${projectDir}/assets/blacklists/hg38-blacklist.bed"
+        }
+        'CHM13' {
+            fasta = "${params.igenomes_base}/Homo_sapiens/UCSC/CHM13/Sequence/WholeGenomeFasta/genome.fa"
```
Comment (FriederikeHanssen) [inline conf/igenomes.config:40]: this alignment of `=` is outragous 😆 🙈

File: modules/local/gtf2bed.nf [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+    path gtf
+
+    output:
+    path '*.bed'       , emit: bed
+    path "versions.yml", emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script: // This script is bundled with the pipeline, in nf-core/rnaseq/bin/
```
Comment (FriederikeHanssen) [inline modules/local/gtf2bed.nf:20]:
```suggestion
    script: // This script is bundled with the pipeline, in nf-core/nascent/bin/
```

File: modules/nf-core/fastp/main.nf [hunk 20]
```diff
@@ -0,0 +1,98 @@
+process FASTP {
+    tag "$meta.id"
+    label 'process_medium'
```
Comment (FriederikeHanssen) [inline modules/nf-core/fastp/main.nf:3]: FYI: i found fastp to be the fastest when running on 12 cpus which is also inline with some issue i found on their repo. NOt sure how critical this is for your datasize, but just in case.

File: nextflow.config [hunk 21]
```diff
@@ -1,150 +1,253 @@
...
 manifest {
-  name = 'nf-core/nascent'
-  author = 'Ignacio Tripodi, Margaret Gruca'
-  homePage = 'https://github.com/nf-core/nascent'
-  description = 'Nascent Transcription Processing Pipeline'
-  mainScript = 'main.nf'
-  nextflowVersion = '>=0.32.0'
-  version = '1.0'
+    name            = 'nf-core/nascent'
+    author          = 'Edmund Miller'
```
Comment (FriederikeHanssen) [inline nextflow.config:210]: same here about the authors, would suggest keeping everyone that was there (again unless you guys agreed on something else :) )

File: nextflow_schema.json [hunk 22]
```diff
@@ -0,0 +1,456 @@
...
+            "description": "",
+            "default": "",
+            "properties": {
+                "aligner": {
+                    "type": "string",
+                    "default": "bwa",
+                    "fa_icon": "fas fa-puzzle-piece",
+                    "enum": ["bwa", "bwamem2", "dragmap"],
+                    "description": "Specify aligner to be used to map reads to reference genome.",
+                    "help_text": "TODO"
```
Comment (FriederikeHanssen) [inline nextflow_schema.json:57]: 😆 no help for you

File: subworkflows/local/align_bwa/main.nf [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+    ch_versions = Channel.empty()
+
+    //
+    // Map reads with BWA MEM
+    //
+    BWA_MEM ( reads, index, true )
+    ch_versions = ch_versions.mix(BWA_MEM.out.versions)
+
+    //
+    // Sort, index BAM file and run samtools stats, flagstat and idxstats
```
Comment (FriederikeHanssen) [inline subworkflows/local/align_bwa/main.nf:24]: are you sorting the bam file twice?

File: subworkflows/local/coverage_graphs.nf [hunk 24]
```diff
@@ -0,0 +1,65 @@
+/*
+ * TODO
+ */
```
Comment (FriederikeHanssen) [inline subworkflows/local/coverage_graphs.nf:3]:
```suggestion
```

File: subworkflows/local/coverage_graphs.nf [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+        fasta,
+        fai
+    )
+
+    DEEPTOOLS_BAMCOVERAGE_MINUS (
+        ch_bam_bai,
+        fasta,
+        fai
+    )
+
```
Comment (FriederikeHanssen) [inline subworkflows/local/coverage_graphs.nf:55]: deeptools versions are not collected

File: subworkflows/local/grohmm.nf [hunk 26]
```diff
@@ -0,0 +1,44 @@
+/*
+ * TODO
+ */
+
```
Comment (FriederikeHanssen) [inline subworkflows/local/grohmm.nf:4]:
```suggestion
```

File: .github/workflows/branch.yml (hunk 1 above)
Comment (Emiller88) [reply .github/workflows/branch.yml:16]: Not that I know of, do you know how to check?

File: subworkflows/local/prepare_genome.nf [hunk 27]
```diff
@@ -0,0 +1,133 @@
...
+    //
+    if (params.fasta.endsWith('.gz')) {
+        ch_fasta = GUNZIP_FASTA ( [ [:], params.fasta ] ).gunzip.map { it[1] }
+        ch_versions = ch_versions.mix(GUNZIP_FASTA.out.versions)
+    } else {
+        ch_fasta = file(params.fasta)
+    }
+
+    // Create Fai file
+    ch_fai = SAMTOOLS_FAIDX( [ [:], ch_fasta ] ).fai.map { it[1] }
```
Comment (FriederikeHanssen) [inline subworkflows/local/prepare_genome.nf:41]: SAMTOOLS_FAIDX version missing

File: subworkflows/local/prepare_genome.nf [hunk 28]
```diff
@@ -0,0 +1,133 @@
...
+        }
+    } else {
+        ch_gene_bed = GTF2BED ( ch_gtf ).bed
+        ch_versions = ch_versions.mix(GTF2BED.out.versions)
+    }
+
+    //
+    // Create chromosome sizes file
+    //
+    ch_chrom_sizes = CUSTOM_GETCHROMSIZES ( [ [:], ch_fasta ] ).sizes
```
Comment (FriederikeHanssen) [inline subworkflows/local/prepare_genome.nf:81]: version missing

File: subworkflows/local/prepare_genome.nf [hunk 29]
```diff
@@ -0,0 +1,133 @@
...
+
+    //
+    // Uncompress BWA index or generate from scratch if required
+    //
+    ch_bwa_index = Channel.empty()
+    ch_dragmap = Channel.empty()
+    if ('bwa' in prepare_tool_indices) {
+        if (params.bwa_index) {
+            if (params.bwa_index.endsWith('.tar.gz')) {
+                ch_bwa_index = UNTAR_BWA_INDEX ( params.bwa_index ).untar
```
Comment (FriederikeHanssen) [inline subworkflows/local/prepare_genome.nf:91]: version missing

File: subworkflows/local/prepare_genome.nf [hunk 30]
```diff
@@ -0,0 +1,133 @@
...
+                ch_bwa_index = file(params.bwa_index)
+            }
+        } else {
+            ch_bwa_index = BWA_INDEX ( ch_fasta ).index
+            ch_versions = ch_versions.mix(BWA_INDEX.out.versions)
+        }
+    } else if ('bwamem2' in prepare_tool_indices) {
+        if (params.bwamem2_index) {
+            if (params.bwamem2_index.endsWith('.tar.gz') || params.bwamem2_index.endsWith('.tgz')) {
+                ch_bwa_index = UNTAR_BWA_INDEX ( [ [:], params.bwamem2_index ] ).untar
```
Comment (FriederikeHanssen) [inline subworkflows/local/prepare_genome.nf:102]: version

File: subworkflows/local/prepare_genome.nf [hunk 31]
```diff
@@ -0,0 +1,133 @@
...
+                ch_bwa_index = file(params.bwamem2_index)
+            }
+        } else {
+            ch_bwa_index = BWAMEM2_INDEX ( [ [:], ch_fasta ] ).index
+            ch_versions = ch_versions.mix(BWAMEM2_INDEX.out.versions)
+        }
+    } else if ('dragmap' in prepare_tool_indices) {
+        if (params.dragmap) {
+            if (params.dragmap.endsWith('.tar.gz')) {
+                ch_bwa_index = UNTAR_DRAGMAP_INDEX ( params.dragmap ).untar
```
Comment (FriederikeHanssen) [inline subworkflows/local/prepare_genome.nf:113]: version

File: subworkflows/local/quality_control.nf [hunk 32]
```diff
@@ -0,0 +1,59 @@
+/*
+ * TODO
+ */
+
```
Comment (FriederikeHanssen) [inline subworkflows/local/quality_control.nf:4]:
```suggestion
```

File: subworkflows/local/quality_control.nf [hunk 33]
```diff
@@ -0,0 +1,59 @@
...
+    RSEQC_READDUPLICATION ( bam )
+    ch_versions = ch_versions.mix(RSEQC_READDUPLICATION.out.versions.first())
+
+    RSEQC_INFEREXPERIMENT (
+        bam,
+        bed
+    )
+    ch_versions = ch_versions.mix(RSEQC_INFEREXPERIMENT.out.versions.first())
+
+    BBMAP_PILEUP ( bam )
```
Comment (FriederikeHanssen) [inline subworkflows/local/quality_control.nf:42]: version

File: subworkflows/local/transcript_identification.nf [hunk 34]
```diff
@@ -0,0 +1,68 @@
...
+    ch_identification_bed = ch_identification_bed.mix(GROHMM.out.bed)
+    ch_versions = ch_versions.mix(GROHMM.out.versions.first())
+
+
+    HOMER_GROSEQ ( group_bams, fasta )
+    ch_identification_bed = ch_identification_bed.mix(HOMER_GROSEQ.out.bed)
+    ch_versions = ch_versions.mix(HOMER_GROSEQ.out.versions.first())
+
+
+    // TODO Merge technical replicates
```
Comment (FriederikeHanssen) [inline subworkflows/local/transcript_identification.nf:33]: should this todo be an issue?

File: tests/config/nextflow.config [hunk 35]
```diff
@@ -0,0 +1,35 @@
...
+    outdir = "output/"
+    publish_dir_mode = "copy"
+    enable_conda = false
+    singularity_pull_docker_container = false
+    max_cpus   = 2
+    max_memory = 6.GB
+    max_time   = 6.h
+}
+
+process {
```
Comment (FriederikeHanssen) [inline tests/config/nextflow.config:11]: feels like some of this could be generalised and go to the template

File: tests/modules/local/grohmm/test.yml (hunk 13 above)
Comment (FriederikeHanssen) [reply tests/modules/local/grohmm/test.yml:3]: 😢

File: tests/subworkflows/local/grohmm/test.yml [hunk 36]
```diff
@@ -0,0 +1,14 @@
+- name: subworkflow grohmm
+  command: nextflow run ./tests/subworkflows/local/grohmm/ -entry test_grohmm -c tests/config/nextflow.config -c tests/subworkflows/local/grohmm/nextflow.config --transcript_identification grohmm
+  # FIXME Tests don't work because of bin/ scripts
+  # tags:
+  # - grohmm
+  # TODO md5 sum these
```
Comment (FriederikeHanssen) [inline tests/subworkflows/local/grohmm/test.yml:6]: sounds like it could be an issue

File: workflows/nascent.nf [hunk 37]
```diff
@@ -0,0 +1,312 @@
...
+    )
+    ch_versions = ch_versions.mix(PREPARE_GENOME.out.versions.first())
+
+    //
+    // SUBWORKFLOW: Read in samplesheet, validate and stage input files
+    //
+    INPUT_CHECK (
+        ch_input
+    )
+
```
Comment (FriederikeHanssen) [inline workflows/nascent.nf:119]: the samplesheet also comes with versions, right?

File: workflows/nascent.nf [hunk 38]
```diff
@@ -0,0 +1,312 @@
...
+    //
+    FASTQC (
+        INPUT_CHECK.out.reads
+    )
+    ch_versions = ch_versions.mix(FASTQC.out.versions.first())
+
+    ch_reads = Channel.empty()
+    if(!params.skip_trimming) {
+        FASTP ( INPUT_CHECK.out.reads, [], [] )
+        ch_reads = FASTP.out.reads
```
Comment (FriederikeHanssen) [inline workflows/nascent.nf:131]: version

**Review by FriederikeHanssen — CHANGES_REQUESTED** [review]
for the versions: sometimes you collect with using `.first()` and sometimes without.

Just some tiny things left from my side:
- some versions are not collected I think. 
- authors possibly back into the manifest
- some of the todo statements seem addressed? or could be issues?

File: bin/parameter_tuning.R (hunk 15 above)
Comment (Emiller88) [reply bin/parameter_tuning.R:2]: Nope, I fixed that TODO awhile!

File: bin/parameter_tuning.R (hunk 16 above)
Comment (Emiller88) [reply bin/parameter_tuning.R:4]: I've worked on it, @sruthipsuresh wrote it originally, so no additional licenses needed.

File: tests/subworkflows/local/grohmm/test.yml (hunk 36 above)
Comment (Emiller88) [reply tests/subworkflows/local/grohmm/test.yml:6]: https://github.com/nextflow-io/nextflow/issues/1868
It is adding them as comments though!

File: conf/igenomes.config (hunk 18 above)
Comment (Emiller88) [reply conf/igenomes.config:40]: I think I just left the mito unaligned out of spite

File: docs/output.md [hunk 39]
```diff
@@ -1,169 +1,388 @@
...
+
+They've also created some bed files that might be useful for analysis.
+
+> We have prepared promoter reference bed files (500 bp regions flanking protein-coding transcripts) for human and mouse genomes:
+
+- [promoter for hg38](https://pints.yulab.org/ref/examples/promoters_1kb_tss_centered.bed.gz): based on GENCODE annotation (v24)
+- [promoter for hg19](https://pints.yulab.org/ref/examples/hg19_promoters_1kb_tss_centered.bed.gz): based on GENCODE annotation (v19)
+- [promoter for mm10](https://pints.yulab.org/ref/examples/mm10_promoters_1kb_tss_centered.bed.gz): based on GENCODE annotation (m23)
+
+## Quatification
```
Comment (FriederikeHanssen) [inline docs/output.md:324]:
```suggestion
## Quantification
```

File: docs/usage.md [hunk 40]
```diff
@@ -1,285 +1,287 @@
...
+
+- If `--genome` is provided then the FASTA and GTF files (and existing indices) will be automatically obtained from AWS-iGenomes unless these have already been downloaded locally in the path specified by `--igenomes_base`.
+- If `--gff` is provided as input then this will be converted to a GTF file, or the latter will be used if both are provided.
+
+## Quantification Options
+
+Currently only featureCounts is supported for quatification. It counts both the genes, and the predicted transcripts.
```
Comment (FriederikeHanssen) [inline docs/usage.md:66]:
```suggestion
Currently only featureCounts is supported for quantification. It counts both the genes, and the predicted transcripts.
```

File: subworkflows/local/align_bwa/main.nf (hunk 23 above)
Comment (Emiller88) [reply subworkflows/local/align_bwa/main.nf:24]: Maybe?

File: tests/config/nextflow.config (hunk 35 above)
Comment (Emiller88) [reply tests/config/nextflow.config:11]: Yeah @maxulysse actually figured that out and added this to the `test.config` of Sarek so it's no longer necessary

```nextflow
// Enable container engines/virtualisation envs for CI testing
// only works when specified with the profile ENV
// otherwise tests can be done with the regular provided profiles
if (System.getenv('PROFILE')) {
    if ("$PROFILE" == "conda") {
        conda.createTimeout                      = "120 min"
        params.enable_conda                      = true
        charliecloud.enabled                     = false
        docker.enabled                           = false
        podman.enabled                           = false
        shifter.enabled                          = false
        singularity.enabled                      = false
    } else if ("$PROFILE" == "docker") {
        docker.enabled                           = true
        docker.userEmulation                     = { params.use_gatk_spark ? false : true }.call()
        charliecloud.enabled                     = false
        podman.enabled                           = false
        shifter.enabled                          = false
        singularity.enabled                      = false
    } else if ("$PROFILE" == "singularity") {
        params.singularity_pull_docker_container = false
        singularity.autoMounts                   = true
        singularity.enabled                      = true
        charliecloud.enabled                     = false
        docker.enabled                           = false
        podman.enabled                           = false
        shifter.enabled                          = false
    }
}
```

Comment (Emiller88) [reply tests/config/nextflow.config:11]: Just going to wait for 2.1.0 https://github.com/nf-core/nascent/issues/84

File: conf/test_copro.config [hunk 41]
```diff
@@ -0,0 +1,44 @@
...
+
+    genome = 'hg38'
+    assay_type = 'CoPRO'
+    filter_bed = "https://pints.yulab.org/ref/examples/promoters_1kb_tss_centered.bed.gz"
+    with_umi = true
+    umitools_dedup_stats = true
+}
+
+process {
+    withName: FASTP {
```
Comment (mahesh-panchal) [inline conf/test_copro.config:28]: A small trick here. Use the full process name in test profiles, to make sure it takes priority over configuration loaded in the `modules.config` when using the simple name there.

File: subworkflows/local/grohmm.nf [hunk 42]
```diff
@@ -0,0 +1,44 @@
...
+    take:
+    bams
+    gtf
+    tuning_file
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    ch_tuning = []
```
Comment (mahesh-panchal) [inline subworkflows/local/grohmm.nf:21]: This seems redundant

Comment (Emiller88) [reply subworkflows/local/grohmm.nf:21]: Ah, it's because I'm using the "sarek-style" so the parameter tuning on runs on:
```nextflow
        ext.when = { !params.skip_grohmm && !params.skip_tuning && params.assay_type == "GROseq" }
```

Comment (mahesh-panchal) [reply subworkflows/local/grohmm.nf:21]: So what actually happens below then? Assignment fails (no exception?) and it retains the `[]` value? Also it should be a rule to mark any processes with a comment if their execution is controlled using `ext.when`.

Comment (Emiller88) [reply subworkflows/local/grohmm.nf:21]: Then it should be handled by an `if` statement, IMO. I was just trying it out, it does make the overall workflow "clean" but this was the exact thing I was afraid of, you have to go dig through the modules.config to see why it didn't execute.

File: workflows/nascent.nf [hunk 43]
```diff
@@ -0,0 +1,314 @@
...
+    ch_multiqc_files = ch_multiqc_files.mix(SUBREAD_FEATURECOUNTS_GENE.out.summary.collect{it[1]}.ifEmpty([]))
+
+    MULTIQC (
+        ch_multiqc_files.collect(),
+        ch_multiqc_config.collect().ifEmpty([]),
+        ch_multiqc_custom_config.collect().ifEmpty([]),
+        ch_multiqc_logo.collect().ifEmpty([])
+    )
+    multiqc_report = MULTIQC.out.report.toList()
+    ch_versions = ch_versions.mix(MULTIQC.out.versions)
```
Comment (mahesh-panchal) [inline workflows/nascent.nf:291]: I need to patch the template. This is redundant.

File: workflows/nascent.nf [hunk 44]
```diff
@@ -0,0 +1,314 @@
...
+    )
+    ch_versions = ch_versions.mix(SUBREAD_FEATURECOUNTS_PREDICTED.out.versions.first())
+
+    SUBREAD_FEATURECOUNTS_GENE (
+        ch_sort_bam.combine(PREPARE_GENOME.out.gtf)
+    )
+    ch_versions = ch_versions.mix(SUBREAD_FEATURECOUNTS_GENE.out.versions.first())
+
+    CUSTOM_DUMPSOFTWAREVERSIONS (
+        ch_versions.unique().collectFile(name: 'collated_versions.yml')
```
Comment (mahesh-panchal) [inline workflows/nascent.nf:253]:
```suggestion
        ch_versions.unique{ it.text }.collectFile(name: 'collated_versions.yml')
```

File: conf/modules.config [hunk 45]
```diff
@@ -0,0 +1,179 @@
...
+            [
+                path: { "${params.outdir}/${params.aligner}/log" },
+                mode: 'copy',
+                pattern: "tab"
+            ]
+        ]
+    }
+
+    withName: 'DRAGMAP_HASHTABLE' {
+        ext.when         = { !params.dragmap && params.aligner == "dragmap" }
```
[the remark this answers is not in the document]
Comment (Emiller88) [reply conf/modules.config:65]: Hmm, I don't know. I think for things like

```
ext.when = { params.save_reference }
```

It's perfect for.

So in this case it doesn't make a lot of sense, but there is an if that handles this. I think some nf-core guidelines on this would be helpful, but I think fixing these are outside the scope of this release.

Made an issue to follow up on this #85 I think coming to some consensus on what makes it easiest to contribute to pipelines. There was also talk that relying on the `ext.when` could easily break in the future with the new config stuff.

File: workflows/nascent.nf (hunk 43 above)
Comment (Emiller88) [reply workflows/nascent.nf:291]: https://github.com/nf-core/tools/issues/1962

File: conf/modules.config [hunk 46]
```diff
@@ -0,0 +1,179 @@
...
+        ext.args = "-checkGC"
+    }
+
+    withName: HOMER_MAKEUCSCFILE {
+        ext.args = "-strand separate"
+    }
+
+    withName: PINTS_CALLER {
+        ext.args = { "--exp-type $params.assay_type" }
+        ext.singularity_pull_docker_container = true
```
Comment (mahesh-panchal) [inline conf/modules.config:162]: Why is this here? Shouldn't this be up to the user?

**Review by mahesh-panchal — APPROVED** [review]

File: conf/modules.config (hunk 46 above)
Comment (Emiller88) [reply conf/modules.config:164]: Because the singularity image doesn't exist on biocontainers, it was a struggle just to get the docker container up there the CI flaked out when they added the package to bioconda.
